dbt/models/marts/candidatures_echelle_locale.sql

```diff
@@ -0,0 +1,31 @@
+select
+    {% if env_var('CI', '') %}
+        id,
+    {% else %}
+        {{ dbt_utils.star(ref('stg_candidatures'), relation_alias='candidatures') }},
+        {{ dbt_utils.star(ref('stg_org_prescripteur'), except=["id_org"], relation_alias='org_prescripteur') }},
+        {{ dbt_utils.star(ref('stg_bassin_emploi'),
+        except=["nom_departement", "nom_region", "type_epci", "id_structure"], relation_alias='bassin_emploi') }},
+        {{ dbt_utils.star(ref('stg_reseaux'), except=["SIRET","id_structure"], relation_alias='rsx') }},
+    {% endif %}
+    nom_org.type_auteur_diagnostic_detaille
+from
+    {{ ref('stg_candidatures') }} as candidatures
+left join {{ ref('stg_bassin_emploi') }} as bassin_emploi
+    on bassin_emploi.id_structure = candidatures.id_structure
+left join {{ ref('stg_emmaus') }} as adherents_emmaus
+    on adherents_emmaus.id_structure = candidatures.id_structure
+left join {{ ref('stg_coorace') }} as adherents_coorace
+    on adherents_coorace.id_structure = candidatures.id_structure
+left join {{ ref('stg_fei') }} as adherents_fei
+    on adherents_fei.id_structure = candidatures.id_structure
+left join {{ ref('stg_unai') }} as adherents_unai
+    on adherents_unai.id_structure = candidatures.id_structure
+left join {{ ref('stg_cocagne') }} as adherents_cocagne
+    on adherents_cocagne.id_structure = candidatures.id_structure
```

Comment:
C'est pas possible d'avoir une table par réseau, il faut qu'on parle. Dejà 5 réseaux, peut-être 20 à terme.

Reply:
Si tu as besoin de créér dynamiquement une colonne pour chaque réseaux existant, il doit y avoir une meilleure solution. Par exemple regarde https://dba.stackexchange.com/questions/311053/how-to-transpose-values-from-one-column-into-columns-with-a-value-from-a-differe

Reply:
proposition : 
- créer 1 seed par fichier d'adhérent (`adherents_cocagne.csv` contient la liste des adhérents cocagne)
- créer 1 staging de tous les adhérents: à partir des seeds, construire une vue staging qui contient une ligne par SIAE et une colonne réseau nous permettant de connaître son réseau
- reposer sur cette unique staging pour le left join dans `candidatures_echelle_locale`

vous en pensez quoi ? à chaque nouvel adhérent on n'a qu'à ajouter la seed et ajouter une ligne dans la staging, on ne touche pas à `candidatures_ech_locale`, je pense même qu'avec dbt on devrait pouvoir créer la staging sur toutes les seeds `adherents_*.csv` avec une `for` loop en jinja, je regarde

Reply:
- 1 seed par réseau
- 1 vue globale (surtout pas 1 vue par réseau)

Reply:
Pas d'accord !
Les seeds sont en public sur le repo et c'est une donnée sensible. Les réseaux ne doivent pas savoir que la structure X au SIRET Y appartient au réseau Z.

Mais l'idée est bonne et c'est pourquoi on fait déjà qque chose de similaire à ce que vous proposez avec ce DAG (https://github.com/gip-inclusion/pilotage-airflow/blob/main/dags/reseau_iae_adherents.py). Les csv sont sur un gdrive, pas accessible au commun des mortels (je dois clarifier ce DAG avec Victor à son retour)

J'avais créé une vue par structure pour pouvoir refaire des extractions ponctuelles aux réseaux, mais c'est ridicule. Si c'est ponctuel je créé un ptit script qui me sort le csv avec leur demandes et voilà.

Donc au final on a bien :
1 seed par réseau (sous forme de ghseet sur gdrive)
1 table reprenant les infos de ces seed  (créée par le DAG)
1 vue globale (`stg_reseaux.sql)` qui contient les info nécessaires pour que tout se fasse avec une jointure et qu'à la fin, sur `candidatures_echelle_locale.sql`, on sache que la structure Z appartient au réseaux XY

Reply:
> Pas d'accord !
©️